CacheManager.StackExchange.Redis.SelfCleaning.Tests/SelfCleaningRedisIntegrationTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using Moq;
+using NUnit.Framework;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning.Tests
+{
+    [TestFixture]
+    public class SelfCleaningRedisIntegrationTests
+    {
+        private const double TIME_TO_LIVE_MILLISECONDS_DIFFERENCE_THRESHOLD = 100;
+
+        private Mock<IServer> _serverMock;
+        private Mock<IDatabase> _databaseMock;
+        private Mock<IConnectionMultiplexer> _connectionMock;
+
+        private IDictionary<RedisKey, (RedisValue[] Values, DateTime InsertionTime)> _fauxDatabase;
+
+        private TimeSpan _timeToLive;
+        private ICacheManager<DummyModel> _cache;
+
+        private ICollection<(CacheItemRemovedEventArgs Args, DateTime Time)> _onRemoveByHandleInvocations;
```

Comment:
1. Im debating between this name and something like "_itemExpiredFromCache".
2. Change the `DateTime Time` field to `DateTime ExtractionFromCacheTime` or `DateTime RemovalFromCacheTime`

Reply:
1. I prefer the current one
2. changed